packages/block-editor/src/components/block-tools/index.js

```diff
@@ -173,8 +186,18 @@ export default function BlockTools( {
 						__unstableContentRef={ __unstableContentRef }
 					/>
 				) }
-				{ ! isZoomOutMode &&
-					( hasFixedToolbar || ! isLargeViewport ) && (
+				{ /* If there is no slot available, such as in the standalone block editor, render within the editor */ }
+				{ isTopToolbarFill && (
```

Comment:
I think instead of adding this new slot and making sure we use it in `edit-site` / `edit-post` ... Why not just render `BlockToolbar` directly in these packages. As mentioned above, this was the original idea and I don't really know when and why that changed.

Reply:
The original PR was also moving the popover block tools into the header, which would remove the `shift+tab` and always have the block tools rendered in the same location in the DOM regardless of mode. This was intended to be an improvement for screen reader and keyboard users, without impacting sighted, mouse users.

I can't recall the exact reason off hand, but I ran into issues with rendering the popover version of the block toolbar within the header. But, we're not doing that in this PR anymore, so this is no longer an issue.

I've been looking at removing the slot and rendering the `<BlockContextualToolbar />` within the post editor header. It's very close to working well.

The only real blocker I'm running into is the floating rich text toolbar that's used for captioning images. It relies on the `__unstableContentRef` passed down from `<BlockTools />`: `const blockToolbarRef = usePopoverScroll( __unstableContentRef );` The inline rich toolbar needs to be in the DOM alongside the block toolbar, and I don't know of a way to access the `__unstableContentRef` from the `<BlockTools />` from within the `edit-post` header.

That said... It doesn't look the `usePopoverScroll` is working on `trunk`. I believe the intention of it is to allow the editor canvas to scroll even if your mouse is on the block toolbar. I'm fine removing this `usePopoverScroll` because...
- **It's not used on the main floating block toolbar.** Only the rich text toolbar and `<Popover.Slot name="__unstable-block-tools-after" ref={ blockToolbarAfterRef } />` use it. And if this functionality was truly warranted, I'd argue the main floating block toolbar should have it.
- **It doesn't work**: The wheel events are firing, but the editor canvas isn't scrolling. It's very possible I'm misunderstanding the intention of the usePopoverScroll though.

I'll push an example of what the code would look like without the Slot implementation sometime today.

Reply:
cc @ellatrix might know more about the inline toolbar but I think it should be fine for this one to be treated differently. Always contextual.

Reply:
There have been conversations around combining the inline toolbar and the contextual toolbar, especially for the top toolbar mode.